openff/qcsubmit/_tests/test_submissions.py

```diff
@@ -835,6 +838,139 @@ def test_optimization_submissions(fulltest_client, specification):
+            ),
+            id="Default conv with maxiter",
+        ),
+        pytest.param(
+            (
+                "CUSTOM",
+                [
+                    "energy",
+                    "1e-4",
+                    "grms",
+                    "3e-2",
+                    "gmax",
+                    "4.5e-1",
+                    "drms",
+                    "1.2e-3",
+                    "dmax",
+                    "1.8e-1",
+                ],
+                300,
+                "Custom convergence, no maxiter",
+            ),
+            id="Custom convergence, no maxiter",
+        ),
+    ],
+)
+def test_optimization_submissions_convergence(fulltest_client, opt_keywords):
+    """Test submitting an Optimization dataset with custom convergence options."""
+
+    client = fulltest_client
+
+    convergence_set, converge, maxit, ds_suffix = opt_keywords
+
+    ethane = Molecule.from_file(get_data("ethane.sdf"), "sdf")
+
+    dataset = OptimizationDataset(
+        dataset_name="Test optimizations with converge " + ds_suffix,
+        description="Test optimization dataset with constraints" + ds_suffix,
+        dataset_tagline="Testing optimization datasets" + ds_suffix,
+    )
+
+    dataset.clear_qcspecs()
+    dataset.add_qc_spec(
+        method="openff-1.0.0",
+        basis="smirnoff",
+        program="openmm",
+        spec_name="test_spec",
+        spec_description="test_spec",
+        overwrite=True,
+    )
+
+    # add the molecule
+    index = ethane.to_smiles()
+    dataset.add_molecule(index=index, molecule=ethane)
+
+    # force a validation error with the GeometricProcedure
+    with pytest.raises(ValidationError):
+        dataset.optimization_procedure = GeometricProcedure(
+            convergence_set="GAU maxiter"
+        )
+    with pytest.raises(ValidationError):
+        dataset.optimization_procedure = GeometricProcedure(converge=["GAU", "maxiter"])
+
+    # re-add the GeometricProcedure so we can submit the data
+    dataset.optimization_procedure = GeometricProcedure(
+        program="geometric",
+        maxiter=maxit,
+        convergence_set=convergence_set,
+        converge=converge,
+    )
+
+    # only save final gradients, results, if --converge maxiter not requested
+    if "maxiter" not in converge:
+        dataset.protocols = OptimizationProtocols(trajectory="initial_and_final")
+
+    # now submit
+    dataset.submit(client=client)
+
+    await_results(
+        client,
+        check_fn=PortalClient.get_optimizations,
+        timeout=240,  # ids = [1, 2]
+    )
+
+    # make sure of the results are complete
+    ds = client.get_dataset(
+        legacy_qcsubmit_ds_type_to_next_qcf_ds_type[dataset.type], dataset.dataset_name
+    )
+
+    query = ds.iterate_records(specification_names="test_spec")
+
+    for name, spec, record in query:
+        assert record.status == RecordStatusEnum.complete
+        assert record.error is None
+
+        assert [
```

Comment:
Nice, so this checks that the custom convergence settings are passed to the record. Could we also check they are picked up by geometric in the stdout of the optimisation?